packages/mongodb-downloader/src/npm-with-lock.ts

```diff
@@ -0,0 +1,406 @@
+// Adapted from:
+// https://raw.githubusercontent.com/npm/cli/072253549d774893a3689341dbc660cb845ebcfe/workspaces/libnpmexec/lib/with-lock.js
+
+// The Artistic License 2.0
```

Comment:
https://docs.google.com/document/d/1esBoY6PP7BID2MLs66isyBq9Gxp6_iZczSxXjYlMW6g/edit?tab=t.0#heading=h.xpnf0sgf7xso 🫠

Reply:
ah, that's a good doc to have on hand. Welp, I guess I'll use [proper-lockfile](https://www.npmjs.com/package/proper-lockfile)